Testing

sbaughman opened the pull request.

sbaughman committed:
Remove unused variable

sbaughman committed:
Remove markTypos and markAllTypos functions

Not used anymore now that all spellchecks are done at block level

sbaughman committed:
Remove more unused variables

sbaughman committed:
Make word walker ignore spellcheck spans

sbaughman committed:
Stop removing all spellcheck spans on every render

sbaughman committed:
Merge branch 'pstat_rewrite' of github.com:PolicyStat/ckeditor-spell-check-plugin-js-dev-edge into spellcheck_spans

sbaughman committed:
Add markTypos back in

not currently used, but will likely be needed in the future.  Still
leaving markAllTypos out, as it is not needed now that everything is
done by block-level

sbaughman committed:
Refactor render function to use markTypos

sbaughman committed:
Move event handler setup so they are all together

sbaughman committed:
Remove outdated comment

sbaughman committed:
refactor spellCheckInProgress

sbaughman committed:
Add checks to wordwalker

before returning an object when ‘WordWalker.getNextWord’ is called,
check to make sure that the word is not an empty string

sbaughman committed:
Add test to verify spellcheck spans are added when a spellcheck occurs

sbaughman committed:
Add test to verify the word walker skips spellcheck spans

sbaughman committed:
Ignoring obsolete tests

‘markAllTypos’ function was removed after rewriting spellchecker to
check by block.  Will find a way to test these cases elsewhere

sbaughman committed:
Add conditions to handle line breaks and other elements that should be word boundaries.

sbaughman committed:
Refactor conditional for 'hitWordBreak'

sbaughman committed:
Add test for properly separating words after breaks, subscripts, and superscripts

sbaughman committed:
Use reset command instead of starting and stopping plugin

sbaughman committed:
Add reset command

caffodian commented on tests/smoke/events.js:
```diff
@@ -57,7 +57,7 @@
 				});
```
idea:  can we move the "resumeAfter" listener registration out of `bot.setData`'s callback?

Probable timing issue where the resume listener is registered after the data has already been set.  I don't know why it is nested here.  Probably misunderstanding on my part, or it used to work, because we were manually starting the spellcheck

sbaughman committed:
Refactor tests to use 'setHtmlWithSelection' instead of 'setData'

having ‘resumeAfter’ in the callback function was causing intermittent
test failures due to timing issues

sbaughman committed:
Remove unneeded delay

sbaughman committed:
Set up for spellcheck tests

sbaughman committed:
Add spellchecking to paths

sbaughman committed:
Move paragraph spellcheck out of events and into new spellchecking test group

sbaughman committed:
Check whole paragraph, not just that span exists

sbaughman committed:
No longer need assertHtml in events tests

sbaughman committed:
rename test

sbaughman committed:
Add ordered list test

sbaughman committed:
Add test for nested lists

sbaughman committed:
Add test for inline style elements

sbaughman committed:
Rename tests

sbaughman committed:
Add test for complex nested list

sbaughman committed:
Add test for simple table

sbaughman committed:
Add test for table after p conversion

sbaughman committed:
Add test for line breaks

sbaughman committed:
Remove non-functional 'markAllTypos' and 'markAllTyposTables' files

sbaughman committed:
Fix typos

sbaughman committed:
Use word walker to get words prior to AJAX call

fixes failing tests for line breaks and complex nested lists

caffodian commented:
I broke github by changing the base branch and it doesn't seem to be fixable.  closing and re-opening.
